logical_switch_external_ids_test.go

```diff
@@ -0,0 +1,124 @@
+/**
```

Comment:
Thanks for adding tests. Could you rename this file to logical_switch_test? I don't think we have one yet.